src/dmd/ctfe/bc.d

```diff
@@ -0,0 +1,2369 @@
+        // anything important
+        byteCodeArray[ip-2] |= (size & 255) << 8;
+
+    }
+
+}
+
+string printInstructions(const uint[] arr, const string[ushort] stackMap = null) pure @trusted
+{
+    return printInstructions(arr.ptr, cast(uint) arr.length, stackMap);
+}
+/*
+string localName(const string[ushort] stackMap, uint addr) pure
+{
+    localName(stackMap, cast(ushort)addr);
+}
+*/
+
+string localName(const string[ushort] stackMap, ushort addr) pure
+{
+    const(string)* name;
+    if (stackMap)
+    {
+        name = addr in stackMap;
+        if (name && *name !is null)
+        {
+            return *name;
+        }
+    }
+
+    return "SP[" ~ itos(addr) ~ "]";
+}
+
+string printInstructions(const uint* startInstructions, uint length, const string[ushort] stackMap = null) pure
+{
+
+    char[] result = cast(char[])"StartInstructionDump: \n";
+    uint pos = 0;
+
+    bool has4ByteOffset;
+    if (length > 4 && startInstructions[0 .. 4] == [0, 0, 0, 0])
+    {
+        has4ByteOffset = true;
+        //length -= 4;
+        //startInstructions += 4;
+        //pos += 4;
+    }
+
+    result ~= "Length : " ~ itos(cast(int)length) ~ "\n";
+    auto arr = startInstructions[0 .. length];
+
+    void printText(uint textLength)
+    {
+        const lengthOverFour = textLength / 4;
+        auto restLength =  textLength % 4;
+        const alignedLength = align4(textLength) / 4;
+
+        // alignLengthBy2
+        assert(alignedLength <= length, "text (" ~ itos(alignedLength) ~") longer then code (" ~ itos(length) ~ ")");
+        auto insertPos = result.length;
+        result.length += textLength;
+
+        result[insertPos .. insertPos + textLength] = '_';
+
+        foreach(chars; arr[pos .. pos + lengthOverFour])
+        {
+            result[insertPos++] = chars >> 0x00 & 0xFF;
+            result[insertPos++] = chars >> 0x08 & 0xFF;
+            result[insertPos++] = chars >> 0x10 & 0xFF;
+            result[insertPos++] = chars >> 0x18 & 0xFF;
+        }
+
+        int shiftAmount = 0;
+        const lastChars = restLength ? arr[pos + lengthOverFour] : 0;
+
+        while(restLength--)
+        {
+            result[insertPos++] = lastChars >> shiftAmount & 0xFF;
+            shiftAmount += 8;
+        }
+
+        pos += alignedLength;
+        length -= alignedLength;
+    }
+
+    while (length--)
+    {
+        uint lw = arr[pos];
+        result ~= itos(pos) ~ ":\t";
+        ++pos;
+        if (lw == 0)
+        {
+            result ~= "0x0 0x0 0x0 0x0\n";
+            continue;
+        }
+
+        // We have a long instruction
+
+        --length;
+        const uint hi = arr[pos];
+        const int imm32c = (*cast(int*)&arr[pos++]);
+
+        final switch (cast(LongInst)(lw & InstMask))
+        {
+        case LongInst.SetHighImm32:
+            {
+                result ~= "SetHigh " ~ localName(stackMap, lw >> 16) ~ ", #" ~ itos(hi) ~ "\n";
```

Comment:
this switch could be less cluttered with a local `string function(string, uint)` `foo("SetHigh ",hi)`